### PR TITLE
Programs

### PR DESCRIPTION
**Pull request by jonathandick**
Added a widget to list programs for which a patient has been enrolled. This widget will allow a user to navigate to program specific dashboards. It will not follow same pattern of Level1, Level2,Level 3 used for other widgets. The designs are pending but have discussed this with Greg Schmidt. 

In addition, I have move put programs next to conditions on the dashboard and moved allergies below them. Note that this ordering is somewhat temporary as in the near future, we will be refactoring patient chart to use a configuration which determines the layout of widgets. 

image

### CODE REVIEW

File: src/summary/history/programs/programs.resource.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { openmrsObservableFetch } from "@openmrs/esm-api";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+
+
+export function performPatientProgramsSearch(
+  patientID: string
+): Observable<PatientPrograms[]> {
+  return openmrsObservableFetch(
+    `/ws/rest/v1/programenrollment?patient=${patientID}`
```
Comment (jonathandick) [inline src/summary/history/programs/programs.resource.tsx]: There is no fhir api for getting program related data. I have created this ticket,
https://issues.openmrs.org/projects/MF/issues/MF-121 to remind us to make the change.

File: src/summary/history/programs/programs.resource.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  ).pipe(
+    map(({ data }) => data["results"]),
+    take(3)
+  );
+}
+
+type PatientPrograms = {
```
Comment (fatmali) [inline src/summary/history/programs/programs.resource.tsx]: Whats the point of this type?

File: .babelrc [hunk 3]
```diff
@@ -6,5 +6,9 @@
     ],
     "plugins": [
```
Comment (jonathandick) [inline .babelrc]: Added the translation plugin using npm i --save-dev babel-plugin-i18next-extract

File: src/summary/history/conditions-card.component.tsx [hunk 4]
```diff
@@ -35,7 +37,7 @@ export default function ConditionsCard(props: ConditionsCardProps) {
   }, [patient]);
 
   return (
-    <SummaryCard name="Conditions" match={props.match}>
+    <SummaryCard name={t("conditions", "Conditions")} match={props.match}>
```
Comment (jonathandick) [inline src/summary/history/conditions-card.component.tsx]: Per @joeldenning's help, you can use this function instead of the <Trans> element when translating a string being passed as a property. 

Note, this probable doesn't belong in this PR but was a test case and such a small amount of code that I'm including it here.

File: src/summary/history/programs/programs-card.component.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  const { t } = useTranslation();
+
+  React.useEffect(() => {
+    const subscription = performPatientProgramsSearch(patientUuid).subscribe(
```
Comment (fatmali) [inline src/summary/history/programs/programs-card.component.tsx]: nit: I think this function could be named simpler, perhaps `getPatientPrograms` or `fetchPatientPrograms`

File: src/summary/history/programs/programs-card.component.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+              color: "var(--omrs-color-ink-medium-contrast)",
+              fontFamily: "Work Sans"
+            }}
+          />
+        </SummaryCardRowContent>
+      </SummaryCardRow>
+      {patientPrograms &&
+        patientPrograms.map(program => {
+          return (
+            <SummaryCardRow key={program.uuid} linkTo="">
```
Comment (fatmali) [inline src/summary/history/programs/programs-card.component.tsx]: why the empty `linkTo`?

Comment (jonathandick) [reply src/summary/history/programs/programs-card.component.tsx]: Because there's no level 2 yet. I will update once I create the level 2.

File: src/summary/history/programs/programs-card.component.tsx (hunk 5 above)
Comment (jonathandick) [reply src/summary/history/programs/programs-card.component.tsx]: Done. Changed to fetchPatientPrograms

File: src/summary/history/programs/programs.resource.tsx (hunk 2 above)
Comment (jonathandick) [reply src/summary/history/programs/programs.resource.tsx]: updated
